File: src/frontend/vue-notus-main/vue-notus-main/vue-notus-main/src/models/TacGia.js
export default class TacGia {
    constructor(data = {}) {
        this.id = data.id || null;
        this.maTacGia = data.maTacGia || '';
        this.tenTacGia = data.tenTacGia || '';
        this.deleted = data.deleted !== undefined ? data.deleted : true;
    }

    toJSON() {
        return {
            id: this.id,
            tenTacGia: this.tenTacGia,
            deleted: this.deleted
        };
    }
} 

File: src/main/java/org/example/datnbbook/model/TacGia.java
package org.example.datnbbook.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.Nationalized;

import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "tac_gia")
public class TacGia {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @ColumnDefault("'TG'+right('-000'+CONVERT([nvarchar](5), NEXT VALUE FOR [dbo].[TGSeq]), 5)")
    @Column(name = "ma_tac_gia", length = 10)
    private String maTacGia;

    @Nationalized
    @Column(name = "ten_tac_gia", length = 100)
    private String tenTacGia;

    @ColumnDefault("0")
    @Column(name = "deleted")
    private Boolean deleted;

//
//    @OneToMany(mappedBy = "idTacGia")
//    private Set<ChiTietSanPham> chiTietSanPhams = new LinkedHashSet<>();

}